spec/models/service_provider_spec.rb

```diff
@@ -38,6 +38,23 @@
 
       expect(service_provider.valid?).to eq(true)
     end
+
+    it 'validates that all redirect_uris are absolute, parsable uris' do
+      valid_sp = build(:service_provider, redirect_uris: ['http://foo.com'])
+      missing_protocol_sp = build(:service_provider, redirect_uris: ['foo.com'])
+      relative_uri_sp = build(:service_provider, redirect_uris: ['/asdf/hjkl'])
+      bad_uri_sp = build(:service_provider, redirect_uris: [' http://foo.com'])
+
+      expect(valid_sp).to be_valid
+      expect(missing_protocol_sp).to_not be_valid
+      expect(relative_uri_sp).to_not be_valid
+      expect(bad_uri_sp).to_not be_valid
```

Comment:
What do you mean? As in do a DNS lookup of the domain? We could maybe do that, but I think that'd fall out of the scope of this PR.

Reply:
It's the only common bad domain pattern I can think of that you don't have covered... the typo that leads to .con instead of .com (just one example). Would a check against permissable values work? Or is it just not worth it?

Reply:
🤔 that may be something to look into, but I think it is a different issue. What I'm doing with this PR is addressing a bug that occurs when we cannot parse the url